PolyEngine/Core/Src/Collections/BitMask.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Dynarray.hpp"
+
+namespace Poly {
+	class CORE_DLLEXPORT BitMask
+	{
+	public:
+		using DataType = unsigned short;
+
+		BitMask(size_t size = sizeof(DataType));
+
+		//Bitwise operators
+		BitMask operator|(const BitMask& rhs) const;
+		BitMask operator^(const BitMask& rhs) const;
+		BitMask operator&(const BitMask& rhs) const;
+		BitMask& operator~();
+		//Bitwise assignment operators
+		BitMask& operator|=(const BitMask& rhs);
+		BitMask& operator^=(const BitMask& rhs);
+		BitMask& operator&=(const BitMask& rhs);
+
+
+		bool operator==(const BitMask rhs) const;
+		bool operator!=(const BitMask rhs) const
+		{
+			return !(*this == rhs);
+		}
+		bool operator[](size_t index) const;
```

Comment:
Please add @todo annotation: "Convert this method to return bit proxy", or if you want I can tell you how to write this proxy. In general we want to be able to do sth like this: `BitMask b; b[0] = true;` which is not possible right now.